main.tf

```diff
@@ -1,28 +1,63 @@
-#######################
+locals {
+  lc_name              = coalesce(var.lc_name, var.name)
+  launch_configuration = var.create_lc ? aws_launch_configuration.this[0].name : var.launch_configuration
+
+  lt_name         = coalesce(var.lt_name, var.name)
+  launch_template = var.create_lt ? aws_launch_template.this[0].name : var.launch_template
+
+  tags = concat(
+    [
+      {
+        "key"                 = "Name"
+        "value"               = var.name
+        "propagate_at_launch" = true
+      },
+    ],
+    var.tags,
+    null_resource.tags_as_list_of_maps.*.triggers,
+  )
+}
+
+resource "null_resource" "tags_as_list_of_maps" {
+  count = length(keys(var.tags_as_map))
+
+  triggers = {
+    "key"                 = keys(var.tags_as_map)[count.index]
+    "value"               = values(var.tags_as_map)[count.index]
+    "propagate_at_launch" = true
+  }
+}
+
+################################################################################
 # Launch configuration
-#######################
+################################################################################
+
 resource "aws_launch_configuration" "this" {
   count = var.create_lc ? 1 : 0
 
-  name_prefix                 = "${coalesce(var.lc_name, var.name)}-"
-  image_id                    = var.image_id
-  instance_type               = var.instance_type
-  iam_instance_profile        = var.iam_instance_profile
-  key_name                    = var.key_name
+  name        = var.lc_use_name_prefix ? null : local.lc_name
```

Comment:
We should be careful with using static `name` because it won't be able to recreate the updated LC/LT resource on update because "name is already in use" (`create_before_destroy = true`).

I think this is the reason why we have only `name_prefix` here.

Reply:
while I agree, it is something that tends to come up. I don't know where it would be applicable honestly, but since the default settings are to set to use the `name_prefix`, I thought it would be safe to add and let users choose a static name if they wish to do so (#76)

Reply:
If a user creates LC with the static name they won't be able to do any changes to any of the properties of this LC. A bit of a surprise for adventurous developers who set `lc_use_name_prefix = false` :)

Ok, we can keep it like this since `name_prefix` is used by default.